api/provider/aws/test_aws/environment_read_test.go

```diff
@@ -0,0 +1,104 @@
+package test_aws
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/golang/mock/gomock"
+	provider "github.com/quintilesims/layer0/api/provider/aws"
+	"github.com/quintilesims/layer0/api/tag"
+	awsc "github.com/quintilesims/layer0/common/aws"
+	"github.com/quintilesims/layer0/common/config/mock_config"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEnvironmentRead(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	// todo: setup helper for config
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	tags := models.Tags{
+		{
+			EntityID:   "env_id",
+			EntityType: "environment",
+			Key:        "name",
+			Value:      "env_name",
+		},
+		{
+			EntityID:   "env_id",
+			EntityType: "environment",
+			Key:        "os",
+			Value:      "linux",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// environments' security group names are <fq environment id>-env
+	describeSecurityGroupHelper(mockAWS, "l0-test-env_id-env", "sg_id")
```

Comment:
since `describeSecurityGroupHelper` is only used for tests, would it be an "improvement" for the function itself to just take `env_id` and concat the rest of the fq name?

Reply:
Doing that would either required we pass in a `instance string` param, or we would be introducing "magic" by assuming `instance`=`"test"`. I wouldn't consider either option an improvement. 

Also, I think it would be confusing to see a mix of `env_id` and `l0-test-env_id` during the "aws mocking" section of the tests.